src/vcpkg/archives.cpp

```diff
@@ -16,22 +16,34 @@ namespace
         (msg::path, msg::exit_code),
         "",
         "msiexec failed while extracting '{path}' with launch or exit code {exit_code} and message:");
+    DECLARE_AND_REGISTER_MESSAGE(
+        CouldNotDeduceNugetIdAndVersion,
+        (msg::path),
+        "",
+        "Could not deduce nuget id and version from filename: {path}");
 
 #if defined(_WIN32)
     void win32_extract_nupkg(const VcpkgPaths& paths, const Path& archive, const Path& to_path)
     {
         const auto nuget_exe = paths.get_tool_exe(Tools::NUGET);
 
-        const auto stem = archive.stem().to_string();
+        const auto stem = archive.stem();
+
         // assuming format of [name].[version in the form d.d.d]
         // This assumption may not always hold
-        std::smatch match;
-        const bool has_match = std::regex_match(stem, match, std::regex{R"###(^(.+)\.(\d+\.\d+\.\d+)$)###"});
-        Checks::check_exit(
-            VCPKG_LINE_INFO, has_match, "Could not deduce nuget id and version from filename: %s", archive);
+        auto dot_after_name = Util::find_nth_from_last(stem, '.', 2);
+
+        auto is_digit_or_dot = [](char ch) {
+            return ch == '.' || Parse::ParserBase::is_ascii_digit(ch);
+        };
+        if (dot_after_name == stem.end() || !std::for_each(dot_after_name, stem.end(), is_digit_or_dot))
```

Comment:
I don't understand why we're testing the return value of `for_each()` here -- did you mean `!std::all_of()`?

Reply:
... yes